### PR TITLE
PR of video stream issue

### PR DESCRIPTION
**Pull request by luwanjia**
1. Add the touch event when click on the screen.
2. Solved the crash problem during video stream transmission.

### CODE REVIEW

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 1]
```diff
@@ -922,11 +922,18 @@ void ConnectionHandlerImpl::OnConnectionEnded(
          session_map.end() != session_it; ++session_it) {
       const uint32_t session_key = KeyFromPair(connection_id, session_it->first);
       const ServiceList &service_list = session_it->second.service_list;
-      for (ServiceList::const_iterator service_it = service_list.begin(), end =
+#ifdef OS_WINCE
+      for (int i = service_list.size() - 1; i >= 0; i--) {
```
Comment (BrandonHe) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @smartdevice475 @luwanjia 
Why not use `cont_iterator` here? such as:
```  
for (ServiceList::const_iterator service_it = service_list.end(),begin =
    service_list.begin(); service_it != begin; --service_it) {
````
Dose WinCE port not support `iterator`?

Comment (BrandonHe) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @smartdevice475 @luwanjia 
Why not use `cont_iterator` here? such as:
```  
for (ServiceList::const_iterator service_it = service_list.end(),begin =
    service_list.begin(); service_it != begin; --service_it) {
````
Dose WinCE port not support `iterator`?

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 2]
```diff
@@ -922,11 +922,18 @@ void ConnectionHandlerImpl::OnConnectionEnded(
...
+        connection_handler_observer_->OnServiceEndedCallback(
+            session_key, service_list[i].service_type, CloseSessionReason::kCommon);
+      }
+#else
+	  for (ServiceList::const_iterator service_it = service_list.begin(), end =
```
Comment (BrandonHe) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @luwanjia @smartdevice475 
Please notice the indent alignment.

**Comment by BrandonHe** [comment]
@luwanjia 
This PR contains mine merge operations, you should firstly use `rebase` command, and then send pull request.

File: src/components/connection_handler/src/connection_handler_impl.cc (hunk 1 above)
Comment (luwanjia) [inline src/components/connection_handler/src/connection_handler_impl.cc]: 支持，但是逆序使用iterator不方便

Comment (luwanjia) [inline src/components/connection_handler/src/connection_handler_impl.cc]: 使用begin与end的道理不一样，begin指向第一个元素，end指向并不是最后一个元素，不能这样用

Comment (luwanjia) [inline src/components/connection_handler/src/connection_handler_impl.cc]: 使用begin与end的道理不一样，begin指向第一个元素，end指向并不是最后一个元素，不能这样用

Comment (luwanjia) [reply src/components/connection_handler/src/connection_handler_impl.cc]: 使用begin与end的道理不一样，begin指向第一个元素，end指向并不是最后一个元素，不能这样用

Comment (luwanjia) [reply src/components/connection_handler/src/connection_handler_impl.cc]: 支持，但是逆序使用iterator不方便

File: src/components/connection_handler/src/connection_handler_impl.cc (hunk 2 above)
Comment (luwanjia) [reply src/components/connection_handler/src/connection_handler_impl.cc]: OK，修改后重发PR

**Comment by luwanjia** [comment]
好的，我先rebase一下

File: src/components/connection_handler/src/connection_handler_impl.cc (hunk 1 above)
Comment (BrandonHe) [reply src/components/connection_handler/src/connection_handler_impl.cc]: @luwanjia 
There have reverse iterator for reverse beginning ```rbegin``` and reverse end```rend```, you can try it.

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 3]
```diff
@@ -922,11 +922,19 @@ void ConnectionHandlerImpl::OnConnectionEnded(
          session_map.end() != session_it; ++session_it) {
       const uint32_t session_key = KeyFromPair(connection_id, session_it->first);
       const ServiceList &service_list = session_it->second.service_list;
+#ifdef OS_WINCE
+      for (ServiceList::const_reverse_iterator service_it = service_list.rbegin();
+        service_it != service_list.rend(); ++service_it) {
```
Comment (BrandonHe) [inline src/components/connection_handler/src/connection_handler_impl.cc]: Wrapped parameters have a 4 space indent

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 4]
```diff
@@ -922,11 +922,19 @@ void ConnectionHandlerImpl::OnConnectionEnded(
...
+        connection_handler_observer_->OnServiceEndedCallback(
```
Comment (BrandonHe) [inline src/components/connection_handler/src/connection_handler_impl.cc:928]: Wrapped parameters have a 4 space indent

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 5]
```diff
@@ -922,11 +922,19 @@ void ConnectionHandlerImpl::OnConnectionEnded(
...
+          session_key, service_it->service_type, CloseSessionReason::kCommon);
```
Comment (BrandonHe) [inline src/components/connection_handler/src/connection_handler_impl.cc]: Wrapped parameters have a 4 space indent

**Comment by BrandonHe** [comment]
This PR is to fix issue #9

**Pull request merged**